File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/eventsourcedentity/EventContext.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.eventsourcedentity;

import kalix.javasdk.MetadataContext;

/** Context for an event. */
public interface EventContext extends EventSourcedEntityContext, MetadataContext {
  /**
   * The sequence number of the current event being processed.
   *
   * @return The sequence number.
   */
  long sequenceNumber();
}

File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/eventsourcedentity/EventContext.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.eventsourcedentity

import kalix.scalasdk.MetadataContext

trait EventContext extends EventSourcedEntityContext with MetadataContext {
  def sequenceNumber: Long
}
